### DISCUSSION TITLE
Shopping list - not related to recipe

### DISCUSSION THREAD

**Discussion by ezpl** [mealie-recipes/mealie#1785] [General]
Hi
I want to use Mealie just as much for shopping lists as for keeping recipes. For this to be efficient, a base selection of groceries should be present, or at least remembered. It seems that unless it is coming from a recipe you can only create "notes" which are not kept for the next time you want to add it to the shopping list.

Is there some smart functionality I am missing or is this just how it is at the present? 

![image](https://user-images.githubusercontent.com/30909027/198686458-cb03570a-9357-4c76-8196-088872d0e5e7.png)

Cheers,

Ezpl

**Comment by ezpl** [comment]
I found something - by clicking the apple-icon it will use a list of foods.
However, when entering something new, it is not saved. 
How are you using this functionality to make it efficient?

**Reply by michael-genson** [reply]
Ah yeah see my reply, just touched on this :)

**Comment by michael-genson** [comment]
You have to toggle using foods, which enables using foods instead of notes:
![image](https://user-images.githubusercontent.com/71845777/198687229-6db414dc-1908-42d5-a6e8-fb8368d7e585.png)

You can add foods and labels without sourcing from a recipe through the manage data section:
`/group/data/foods`

If you want a set of "base" foods, you can seed food data from that page as well.

**Reply by ezpl** [reply]
Thanks. Not quite obvious to find it under Groups. Is it possible to update the base food list or seeding form an external source?

**Reply by michael-genson** [reply]
Yup! On that same page if you scroll to the bottom you can seed a base set of foods:
![image](https://user-images.githubusercontent.com/71845777/198688251-8d73751c-d156-48c2-bbbe-a038e32e138a.png)

Note that if you already have foods in your database, and the seeded data has the same foods, they will be added as duplicates so you'll need to sort those out manually from the data menu.

**Reply by michael-genson** [reply]
> Not quite obvious to find it under Groups.

Yeah, maybe we can add some more controls to the shopping list page.

**Reply by ezpl** [reply]
Starting to look useful :-)

So, I did an import in Norwegian, but as it was not (fully) translated everything is now in English and has no labels. 
Is there a way to flush everything and create my own seed list?

**Reply by michael-genson** [reply]
Ah, unfortunately there's no way to delete all food data as far as I'm aware. If you're comfortable with a programming language you can delete foods through the API (or if you're comfortable enough with Python to run a script I can send you some code). Otherwise I believe you need to delete one at a time.

Similarly for creation, you can create one food at a time via the UI, or import via the API if you have the list of foods somewhere else.

**Reply by ezpl** [reply]
OK, the bulk management I can do through the API (or please do share the script - then I'll figure out the easiest route - working normally with c#, not Python, but whatever) 😎
However, the showstopper here would probably be my wife if we tried using this as our common shopping list when she cannot just add "my-random-never-heard-of-grocery" to the built-in list 😏

**Reply by ezpl** [reply]
...or I could create a super-simplified dashboard component in HomeAssistant using the API.

**Reply by michael-genson** [reply]
> (or please do share the script - then I'll figure out the easiest route - working normally with c#, not Python, but whatever)

You'll probably have an easier time with it yourself then, it's a pretty simple REST API. Head to `yourdomain.com/docs` to check out the API.

> the showstopper here would probably be my wife if we tried using this as our common shopping list when she cannot just add "my-random-never-heard-of-grocery" to the built-in list

Yeah I have mine hooked up to Alexa/AWS so a Lambda function does the lookup with fuzzy matching. For a similar reason :)

**Reply by ezpl** [reply]
Will have a look at the API 👍
Voice control could be an even better way to go 😎
The lambda for posting to the API for Alexa could be somehow reusable I would assume with a different input method - would you want to share to give me a head start? 😊

**Reply by michael-genson** [reply]
Specifically with Alexa it's... complicated. If you want to integrate with the Alexa shopping list, you have to create a custom Alexa skill that can be triggered out of session:
https://developer.amazon.com/en-US/docs/alexa/custom-skills/get-a-user-specific-access-token.html

I'm working on making a "shopping list integration app" that does the heavy lifting for developers, but it's a WIP.

Another alternative is to bypass the Alexa shopping list entirely and write a custom skill that calls the Mealie API. So, for instance, you can say "Alexa, ask _my Alexa skill_ to add _something_ to the shopping list." It's still quite a pain since you need a custom Alexa skill in AWS.

The easiest solution, which is the one I'm using currently, is to integrate your shopping list through a different list app that supports Alexa integration (I use ToDoist). Unfortunately you lose a lot of the Mealie functionality doing this, hence the app I'm working on.

---

If you're not tied to Alexa, setting up home assistant with voice controls would be much, _much_ easier.

**Reply by ezpl** [reply]
I will stay away from Alexa 😁

Seems like the /docs section is unavaliable - wonder if it is a bug in the current release? Even on the demo site 😏
https://mealie-demo.hay-kot.dev/docs
